yew-ui/src/components/attendants.rs

```diff
@@ -208,14 +255,88 @@ impl Component for AttendantsComponent {
                     true
                 }
             },
-            Msg::OnPeerAdded(_email) => true,
+            Msg::OnPeerAdded(_email) => {
+                log!("New peer arrived.");
+                if self.e2ee_enabled {
+                    self.send_public_key(ctx);
+                }
+                true
+            }
             Msg::OnFirstFrame((_email, media_type)) => match media_type {
                 MediaType::SCREEN => true,
                 _ => false,
             },
             Msg::OnInboundMedia(response) => {
-                if let Err(e) = self.peer_decode_manager.decode(response) {
-                    log!("error decoding packet: {:?}", e);
+                match response.packet_type.enum_value() {
+                    Ok(PacketType::AES_KEY) => {
+                        if !self.e2ee_enabled {
+                            return false;
+                        }
+                        log!("Received AES_KEY", &response.email);
+                        if let Ok(bytes) = self.rsa.decrypt(&response.data) {
+                            let aes_packet = AesPacket::parse_from_bytes(&bytes).unwrap();
+                            self.peer_keys.insert(
+                                response.email,
+                                Aes128State::from_vecs(
+                                    aes_packet.key,
+                                    aes_packet.iv,
+                                    self.e2ee_enabled,
+                                ),
+                            );
+                        }
+                        return false;
+                    }
+                    Ok(PacketType::RSA_PUB_KEY) => {
+                        if !self.e2ee_enabled {
+                            return false;
+                        }
+                        log!("Received RSA_PUB_KEY");
+                        let rsa_packet = RsaPacket::parse_from_bytes(&response.data).unwrap();
```

Comment:
these unwraps might cause an issue down the line, is there a way to use `and_then` and log the errors if any?